lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -2349,28 +2436,132 @@ void HdVP2Material::_UpdateShaderInstance(const HdMaterialNetwork& mat)
 
 /*! \brief  Acquires a texture for the given image path.
  */
-const HdVP2TextureInfo&
-HdVP2Material::_AcquireTexture(const std::string& path, const HdMaterialNode& node)
+const HdVP2TextureInfo& HdVP2Material::_AcquireTexture(
+    HdSceneDelegate*      sceneDelegate,
+    const std::string&    path,
+    const HdMaterialNode& node)
 {
     const auto it = _textureMap.find(path);
     if (it != _textureMap.end()) {
         return it->second;
     }
 
-    bool                 isSRGB = false;
-    MFloatArray          uvScaleOffset;
-    MHWRender::MTexture* texture = _LoadTexture(path, isSRGB, uvScaleOffset, node);
+    if (_IsDisabledAsyncTextureLoading()) {
+        bool        isSRGB = false;
+        MFloatArray uvScaleOffset;
+
+        VtValue value;
+        auto    fallbackIt = node.parameters.find(_tokens->fallback);
+        if (fallbackIt != node.parameters.end()) {
+            value = fallbackIt->second;
+        }
+
+        MHWRender::MTexture* texture = _LoadTexture(path, isSRGB, uvScaleOffset, value);
+
+        HdVP2TextureInfo& info = _textureMap[path];
+        info._texture.reset(texture);
+        info._isColorSpaceSRGB = isSRGB;
+        if (uvScaleOffset.length() > 0) {
+            TF_VERIFY(uvScaleOffset.length() == 4);
+            info._stScale.Set(
+                uvScaleOffset[0], uvScaleOffset[1]); // The first 2 elements are the scale
+            info._stOffset.Set(
+                uvScaleOffset[2], uvScaleOffset[3]); // The next two elements are the offset
+        }
+
+        return info;
+    }
+
+    VtValue fallbackValue;
+    auto    fallbackIt = node.parameters.find(_tokens->fallback);
+    if (fallbackIt != node.parameters.end()) {
+        fallbackValue = fallbackIt->second;
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(_textureLoadingMutex);
+        _textureLoadingTasks.emplace(
+            path, new TextureLoadingTask(this, sceneDelegate, path, fallbackValue));
+    }
+
+    static HdVP2TextureInfo dummyInfo;
```

Comment:
This static variable is weird. We make a singleton dummyInfo which is grey, except if the first node that needed it happened to have a fallback color, in which case we use that nodes fallback color for everyone?

Reply:
I changed this one a little bit: if there is no fallback color, all the materials would reuse the same "grey" `HdVP2TextureInfo`; otherwise create a `HdVP2TextureInfo` with their own fallback color respectively.
Hope that makes more sense!